src/IW/Sync/Update.hs

```diff
@@ -14,27 +14,36 @@ import IW.App (WithError)
 import IW.Core.Repo (Repo (..))
 import IW.Db (WithDb, upsertRepos, updateRepoCategories)
 import IW.Effects.Cabal (MonadCabal (..), getCabalCategories)
-import IW.Sync.Search (fetchAllHaskellRepos, fromGitHubRepo)
+import IW.Sync.Search (searchAllHaskellRepos)
 
 
--- | This function fetches the latest repos from the GitHub API, parses their @.cabal@ files,
--- and upserts them into the database.
-fetchAndUpsertRepos
+-- | This function fetches repos from the GitHub API within a specified date range,
+-- parses their @.cabal@ files, and upserts them into the database.
+syncRepos
     :: forall env m.
        ( MonadCabal m
        , MonadUnliftIO m
        , WithDb env m
        , WithLog env m
        , WithError m
        )
-    => m ()
-fetchAndUpsertRepos = do
-    gitHubRepos <- fetchAllHaskellRepos
-    let repos = map fromGitHubRepo gitHubRepos
+    => Integer -- ^ Starting interval
+    -> m ()
+syncRepos interval = do
```

Comment:
Is this `interval` a step for syncing?

Reply:
@chshersh This interval is used for the search functions. The interval is for determining the date interval. Look at the search functions for a better understanding, but basically it stands for many days worth of issues/repos we’re fetching. The comments on some of the functions need to be updated. I’m doing that rn.